File: frontend/src/types/study.ts
export interface Flashcard {
  id: string;
  topic: string;
  front: string;
  back: string;
}

export interface QuizQuestion {
  id: string;
  topic: string;
  question: string;
  options: string[];
  correct_index: number;
  explanation: string;
}

export interface TopicSummary {
  name: string;
  key_points: string[];
}

export interface StudyKit {
  title: string;
  summary: string;
  topics: TopicSummary[];
  flashcards: Flashcard[];
  quiz: QuizQuestion[];
}

File: backend/app/schemas/study.py
from pydantic import BaseModel

class Flashcard(BaseModel):
    id: str
    topic: str
    front: str
    back: str

class QuizQuestion(BaseModel):
    id: str
    topic: str
    question: str
    options: list[str]
    correct_index: int
    explanation: str

class TopicSummary(BaseModel):
    name: str
    key_points: list[str]

class StudyKit(BaseModel):
    title: str
    summary: str
    topics: list[TopicSummary]
    flashcards: list[Flashcard]
    quiz: list[QuizQuestion]
